cdap-api/src/main/java/co/cask/cdap/api/schedule/ProgramStatusTriggerInfo.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import co.cask.cdap.api.ProgramStatus;
+import co.cask.cdap.api.app.ApplicationSpecification;
+import co.cask.cdap.api.app.ProgramType;
+import co.cask.cdap.api.workflow.WorkflowToken;
+
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * The program status trigger information to be passed to the triggered program.
+ */
+public class ProgramStatusTriggerInfo extends TriggerInfo {
+
+  private final String namespace;
+  private final ApplicationSpecification applicationSpecification;
+  private final ProgramType programType;
+  private final String program;
+  private final Set<ProgramStatus> triggerStatues;
+  @Nullable
```

Comment:
These fields are nullable because of composite trigger. For instance, if the composite trigger is an OR trigger which contains two program status trigger, and only one of them is satisfied with a program run, then the unsatisfied one does not have runId or workflow token